### PR TITLE
Removed curl from base functionality

### PR DESCRIPTION
**Pull request by Michael-Burke**
- Removed curl and replaced with wget
- Updated package dependencies

### CODE REVIEW

File: src/_setup_packages.sh [hunk 1]
```diff
@@ -76,7 +78,7 @@ function install_docker() {
       "apt")
         info "installing docker, this might take some time..."
         _system_cmd_with_debug_and_fail "mkdir -p /etc/apt/keyrings; \
-          curl -fsSL https://download.docker.com/linux/ubuntu/gpg | \
+          wget --output-document - --quiet https://download.docker.com/linux/ubuntu/gpg | \
```
Comment (mmorgan-pt) [inline src/_setup_packages.sh:81]: maybe use shorthand -O and -q like elsewhere?

File: src/_check.sh [hunk 2]
```diff
@@ -31,7 +31,7 @@ function mod_check() {
 }
 function check_for_maintenance_mode() {
   title "Checking Maintenance Mode"
-  IN_MAINTENANCE=$(curl -ks https://localhost/api/v2/health/full | jq .data.inMaintenanceMode) || IN_MAINTENANCE="Unknown"
+  IN_MAINTENANCE=$(wget -O - -q https://localhost/api/v2/health/full | jq .data.inMaintenanceMode) || IN_MAINTENANCE="Unknown"
```
[the remark this answers is not in the document]
Comment (mmorgan-pt) [reply src/_check.sh:34]: when doing plextrac update, I got the following error that I think is related to this line
```
[+] Waiting up to 90s for application startup
[+] ................ Done
jq: error (at <stdin>:1): Cannot index string with string "Health"
jq: error (at <stdin>:2): Cannot index string with string "Health"
jq: error (at <stdin>:3): Cannot index string with string "Health"
jq: error (at <stdin>:4): Cannot index string with string "Health"
jq: error (at <stdin>:5): Cannot index string with string "Health"
jq: error (at <stdin>:6): Cannot index string with string "Health"
jq: error (at <stdin>:7): Cannot index string with string "Health"
jq: error (at <stdin>:8): Cannot index string with string "Health"
jq: error (at <stdin>:9): Cannot index string with string "Health"
jq: error (at <stdin>:10): Cannot index string with string "Health"
    Try running with '-v' for verbose output for more details
    Try running with '-v' for verbose output for more details
    Try running with '-v' for verbose output for more details
```

File: src/_update.sh [hunk 3]
```diff
@@ -107,8 +107,9 @@ function selfupdate_doUpgrade() {
   debug "Tempdir: $tempDir"
   target="${PLEXTRAC_HOME}/.local/bin/plextrac"
 
-  debug "`curl -w %{url_effective} -L -o $tempDir/$(jq '.name, " ", .browser_download_url' -r <<<$scriptAsset) 2>&1 || error "Release download failed"`"
-  debug "`curl -w %{url_effective} -L -o $tempDir/$(jq '.name, " ", .browser_download_url' -r <<<$scriptAssetSHA256SUM) 2>&1 || error "Checksum download failed"`"
+  debug "`wget $releaseApiUrl -O $tempDir/$(jq '.[].name, " ", .browser_download_url' -r <<<$scriptAsset) 2>&1 || error "Release download failed"`"
```
Comment (mmorgan-pt) [inline src/_update.sh:110]: got error 
```
jq: error (at <stdin>:34): Cannot index string with string "name"
```
within vagrant environment

**Review by mmorgan-pt — CHANGES_REQUESTED** [review]

File: src/_check.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (Michael-Burke) [reply src/_check.sh:34]: Adjusted this to match expected output. Added the `--no-check-certificate` value to the wget command

File: src/_setup_packages.sh (hunk 1 above)
Comment (Michael-Burke) [reply src/_setup_packages.sh:81]: Done

File: src/_update.sh (hunk 3 above)
Comment (Michael-Burke) [reply src/_update.sh:111]: Still looking at this one. It was working when I originally created the PR :thinking:

Comment (Michael-Burke) [reply src/_update.sh]: This should be corrected now; needed `jq -re  '.name'...` and NOT `jq -re '.[].name'...` as it was too expanded to find the expected value

File: src/_check.sh [hunk 4]
```diff
@@ -31,7 +31,7 @@ function mod_check() {
 }
 function check_for_maintenance_mode() {
   title "Checking Maintenance Mode"
-  IN_MAINTENANCE=$(curl -ks https://localhost/api/v2/health/full | jq .data.inMaintenanceMode) || IN_MAINTENANCE="Unknown"
+  IN_MAINTENANCE=$(wget -O - -q https://127.0.0.1/api/v2/health/full | jq .data.inMaintenanceMode --no-check-certificate) || IN_MAINTENANCE="Unknown"
```
Comment (mmorgan-pt) [inline src/_check.sh:34]:
```suggestion
  IN_MAINTENANCE=$(wget -O - -q --no-check-certificate https://127.0.0.1/api/v2/health/full | jq .data.inMaintenanceMode) || IN_MAINTENANCE="Unknown"
```

Comment (mmorgan-pt) [reply src/_check.sh:34]: throws jq errors with the current position of this flag :-)

**Review by mmorgan-pt — CHANGES_REQUESTED** [review]

File: src/_check.sh (hunk 4 above)
Comment (Michael-Burke) [reply src/_check.sh]: Whooops! @mmorgan-pt should be fixed now

File: src/_update.sh [hunk 5]
```diff
@@ -46,6 +46,8 @@ function mod_update() {
   done
 
   mod_check
+
+  title "Update complete"
```
Comment (Michael-Burke) [inline src/_update.sh:50]: Added this to show the update as complete

**Review by mmorgan-pt — APPROVED** [review]
Tested these changes and it all looks great!
Nice work!

**Pull request merged**